src/client.ts

```diff
@@ -343,23 +353,46 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
   }
 
   public async fetchQualifiedSegments(options?: optimizely.OptimizelySegmentOption[]): Promise<boolean> {
-    if (!this.userContext || !this.isReady()) {
+    if (this.odpExplicitlyOff) {
+      return true;
+    }
+
+    if (!this.userContext) {
```

Comment:
Do we need to keep track of "odpOff" here in react-sdk? It'll be tracked and returned properly in JS core.

Reply:
The bool returned from this method, means there were no errors.  

If ODP is expressly off, this method should assume that there were no errors and return `true` regardless if the `userContext` is truthy or not.

True, the underlying JS SDK runs this check earlier during its createInstance and sets the ODP stack to `undefined`. Based on this, the [JS client returns](https://github.com/optimizely/javascript-sdk/blob/master/lib/optimizely/index.ts#L1767)  `null` which is [interpreted by the userContext as a failure](https://github.com/optimizely/javascript-sdk/blob/master/lib/optimizely_user_context/index.ts#L261) which is interpreted by the React SDK that the user is not ready.